src/main/java/org/opensearch/sdk/authz/ProtectedRoute.java

```diff
@@ -0,0 +1,71 @@
+package org.opensearch.sdk.authz;
+
+import org.opensearch.rest.RestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.sdk.ExtensionRestResponse;
+import org.opensearch.sdk.handlers.ExtensionRouteRequestHandler;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ProtectedRoute extends RestHandler.Route {
+
+    private static final Pattern ROUTE_PARAM_REGEX = Pattern.compile("\\{(.*?)\\}");
+
+    public ExtensionRouteRequestHandler requestHandler;
+
+    private String requiredPermission;
+    public ProtectedRoute(RestRequest.Method method, String path, String requiredPermission, ExtensionRouteRequestHandler requestHandler) {
+        super(method, path);
+        this.requestHandler = requestHandler;
+        this.requiredPermission = requiredPermission;
+    }
+
+    public String getRequiredPermission() {
+        return this.requiredPermission;
+    }
+
+    public Pattern getRouteRegex() {
+        if (getPath() == null) {
+            return null;
+        }
+        String routeRegex = getPath().replaceAll("\\{.*?\\}", "(.*)");
+        Pattern routePattern = Pattern.compile(routeRegex);
+        return routePattern;
+    }
+
+    public Map<String, Object> getConsumedParams(String uri) {
+        // Extracting only route and URI params for now. Needs to extract params from the body when that is available
+        Map<String, Object> consumedParams = new HashMap<>();
+        Matcher nameMatcher = ROUTE_PARAM_REGEX.matcher(getPath());
```

Comment:
When messing around with replacing wildcard parameters in routes, which are Paths, I thought that glob matching would be an easier implementation than regex.  See https://docs.oracle.com/javase/8/docs/api/java/nio/file/PathMatcher.html